elastic/kibana
# Reschedule a rule task when there is a cluster_block_exception

Resolves: https://github.com/elastic/response-ops-team/issues/249

This PR reschedules a rule (its task behind it) to be executed again in 1m in case of `cluster_block_exception` that will be thrown during a reindex.

## To verify:

- Run your local Kibana,
- Create a user with `kibana_system` and `kibana_admin` roles
- Logout and login with your new user
- Create an Always firing rule with 5m interval
- Use below request to put a write block on the index used by the Always firing rule.
   `PUT /.internal.alerts-default.alerts-default-000001/_block/write`
- There shouldn't be any error on the terminal
- Check the task SO of the rule by using the below query, value of the `task.schedule.interval` should be 1m and the value of the `task.state` should remain the same.
- Use below request on the Kibana console to remove write block. Every thing should come back to normal and alerts of the rule should be saved under `.internal.alerts-default.alerts-default-000001` index.
```
PUT /.kibana_task_manager_9.0.0_001/_settings
{
  "index": {
    "blocks.write": false
  }
}
```
